helper/jfr.go

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2018-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package helper
+
+import (
+	"os"
+
+	"github.com/paketo-buildpacks/libpak/sherpa"
+
+	"github.com/paketo-buildpacks/libpak/bard"
+)
+
+type JFR struct {
+	Logger bard.Logger
+}
+
+func (j JFR) Execute() (map[string]string, error) {
+	if val, ok := os.LookupEnv("BPL_JAVA_FLIGHT_RECORDER_ENABLED"); !ok || val != "true" {
+		return nil, nil
+	}
+
+	argList := sherpa.GetEnvWithDefault("BPL_JFR_ARGS", "")
+
+	j.Logger.Infof("Enabling Java Flight Recorder")
+
+	// minimum flag to enable JFR, with default config args
+	jfrConfig := "-XX:StartFlightRecording="
```

Comment:
If you do not specify any `BPL_JFR_ARGS`, the default behavior seems to require a user to connect with `jcmd` or remotely with JMC to get a recording. 

I'm trying to think of some default we could use that might be easier to consume without further interaction 🤔 

I had a few thoughts, but the simplest one would seem to be just dumping on exit. We can enable the `dumponexit=true` option. This would trigger a flight recording to be dumped on exit to the `/workspace` directory. We would probably also need to set `filename=/tmp/` so that it writes to the temp directory which would be writable for a non-cnb user, although perhaps not? Perhaps that's up to the user if the user chooses to not be the cnb user.

   At any rate, this would result in JFR running and dumping on exit if there is a problem, similar to what we do with Java NMT. I don't believe this writes anything to disk, except on exit, so that keeps disk usage and I/O low.

tl;dr if `BPL_JFR_ARGS` is empty, then we should set `dumponexit=true` to the default. I think it would be sufficient to either get default or fully custom (i.e. don't try merging user settings and the defaults).